state-chain/pallets/cf-broadcast/src/tests.rs

```diff
@@ -49,45 +54,64 @@ thread_local! {
 // When calling on_idle, we should broadcast everything with this excess weight.
 const LARGE_EXCESS_WEIGHT: Weight = Weight::from_parts(20_000_000_000, 0);
 
+type ValidatorId = <Test as Chainflip>::ValidatorId;
+
 struct MockCfe;
 
 impl MockCfe {
 	fn respond(scenario: Scenario) {
+		// Process non-cfe events (move this out of MockCfe?)
 		let events = System::events();
 		System::reset_events();
 		for event_record in events {
-			Self::process_event(event_record.event, scenario.clone());
+			Self::process_event(event_record.event);
+		}
+
+		// Process cfe events
+		let events = MockCfeEventEmitter::take_events();
+		for event in events {
+			Self::process_cfe_event(event, scenario);
+		}
+	}
+
+	fn process_cfe_event(event: MockCfeEvent<ValidatorId>, scenario: Scenario) {
+		match event {
+			MockCfeEvent::EthTxBroadcastRequest(TxBroadcastRequest {
+				broadcast_id,
+				nominee,
+				payload: _,
+			}) => {
+				match scenario {
+					Scenario::BroadcastFailure => {
+						assert_ok!(Broadcaster::transaction_failed(
+							RawOrigin::Signed(nominee).into(),
+							broadcast_id,
+						));
+					},
+					Scenario::Timeout => {
+						// Ignore the request.
+					},
+					_ => {
+						// only nominee can return the signed tx
+						assert_eq!(
+							nominee,
+							MockNominator::get_last_nominee().unwrap(),
+							"CFE using wrong nomination"
+						);
+					},
+				}
+			},
+			_ => {
+				// No other events used in these tests
+			},
 		}
 	}
 
-	fn process_event(event: RuntimeEvent, scenario: Scenario) {
+	fn process_event(event: RuntimeEvent) {
```

Comment:
I think now we can remove all the events from here - except the timedout case of course. Just to remove some clutter - it's clear which events are required by the CFE now